var/spack/repos/builtin/packages/phist/package.py

```diff
@@ -0,0 +1,136 @@
+    version('master',
+            git='https://bitbucket.org/essex/phist/phist.git', branch='master')
+    version('1.4.1', '53ca2c2c000a36790e1626ed1eb642e3',
+            url='https://bitbucket.org/essex/phist/get/phist-1.4.1.tar.gz')
+
+    # note: there is no virtual package for lapacke (the C bindings for lapack),
+    # We look for a file like lapacke.h in the lapack prefix, if it is found we
+    # pass it to phist's cmake, otherwise we hope that phist finds it itself.
+    def lapacke_include_dir(self):
+        prefix=self.spec['lapack'].prefix
+        include_dir="NOTFOUND"
+        for root, dir, files in os.walk(prefix):
+            if fnmatch.filter(files, "*lapacke.h"):
+                include_dir=root
+                break
+        return include_dir
+
+    def cmake_args(self):
+        spec=self.spec
+        outlev=spec.variants['outlev'].value
+        lapack_libs = (spec['lapack'].libs + spec['blas'].libs).joined(';')
+        lapacke_inc = self.lapacke_include_dir()
+        tpl_lapacke_include_dirs=''
+        if lapacke_inc != "NOTFOUND":
+            tpl_lapacke_include_dirs='-DTPL_LAPACKE_INCLUDE_DIRS='+lapacke_inc
+        # Use everything until the first '+' sign as the kernel library.
+        # Note that we already restrict the string to a list of given values,
+        # so we don't need to check more carefully here.
+        kernel_lib=spec.variants['kernel_lib'].value.split('+',1)[0]
+                    
+        args = ['-DPHIST_KERNEL_LIB=%s' % kernel_lib,
+                '-DPHIST_OUTLEV=%s' % outlev,
+                '-DTPL_LAPACKE_LIBRARIES=%s' % lapack_libs,
+                tpl_lapacke_include_dirs,
+                '-DPHIST_ENABLE_MPI:BOOL=%s' % ('ON' if '+mpi' in spec else 'OFF'),
+                '-DBUILD_SHARED_LIBS:BOOL=%s' % ('ON' if '+shared' in spec else 'OFF'),
+                '-DPHIST_USE_TRILINOS_TPLS:BOOL=%s' % ('ON' if '+trilinos' in spec else 'OFF'),
+                '-DPHIST_USE_SOLVER_TPLS:BOOL=%s' % ('ON' if '+trilinos' in spec else 'OFF'),
+                '-DPHIST_USE_PRECON_TPLS:BOOL=%s' % ('ON' if '+trilinos' in spec else 'OFF'),
+                ];
+                        
+        return args
+
+    @run_after('build')
+    @on_package_attributes(run_tests=True)
+    def check(self):
+        with working_dir(self.build_directory):
+            make("check")
+
+
+    @run_after('install')
+    @on_package_attributes(smoke_test=True)
+    def test_install(self):
+        with working_dir(self.build_directory):
+            make("test_install")
+
+    variant(name='kernel_lib',   default='builtin',
+            description='select the kernel library (backend) for phist',
+            values=['builtin','epetra','tpetra','petsc','petsc+complex','eigen','ghost'])
+    variant(name='outlev', default='2',            description='verbosity. 0: errors 1: +warnings 2: +info 3: +verbose 4: +extreme 5; +debug')
```

Comment:
you can set `values=` to make sure only supported ones are provided (same as `kernel_lib` above)